orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -2543,6 +2545,23 @@ void GradientCheckerMinMaxGradHelper(const std::string op) {
     ASSERT_STATUS_OK(gradient_checker.ComputeGradientError(op_def, {x1_info, x2_info}, {y_info}, &max_error));
     EXPECT_IS_TINY(max_error);
   }
+
+  // More than 2 inputs.
+  {
+    TensorInfo x2_info({2, 3}, true, &x2_transformer);
+    TensorInfo x3_info({2, 3}, true, &x3_transformer);
+    ASSERT_STATUS_OK(gradient_checker.ComputeGradientError(op_def, {x1_info, x2_info, x3_info}, {y_info}, &max_error));
+    EXPECT_IS_TINY(max_error);
+  }
+
+  {
+    TensorInfo x2_info({3}, true, &x2_transformer);
+    TensorInfo x3_info({2, 1}, true, &x3_transformer);
+    TensorInfo x4_info({2, 3}, true, &x4_transformer);
+    ASSERT_STATUS_OK(
+        gradient_checker.ComputeGradientError(op_def, {x1_info, x2_info, x3_info, x4_info}, {y_info}, &max_error));
+    EXPECT_IS_TINY(max_error);
+  }
 }
```

Comment:
Shall we add a case like this max(a, a, a)

Reply:
As the comment above data transformer mentioned, NumericJacobian's calculation logic will fail to handle non-derivable points, that's why we add such data transformers to guarentee no equal values. If your case max(a, a, a) means all inputs are same input tensor, then y = max(a,a,a) are same as y=a, in such case, the NumericJacobian might work, but this is not a valid case logically, just don't know why a model writer put this code in his code, but not just use y=a.